apps/application-system/api/src/app/modules/application/charge/application-charge.service.ts

```diff
@@ -17,21 +18,44 @@ export class ApplicationChargeService {
     this.logger = logger.child({ context: 'ApplicationChargeService' })
   }
 
-  async deleteCharge(application: Pick<Application, 'id' | 'externalData'>) {
+  async deleteCharge(
+    application: Pick<Application, 'id' | 'externalData' | 'typeId' | 'state'>,
```

Comment:
Should we remove external data here and use the Id from `findPaymentByApplicationId`.